### DISCUSSION TITLE
Add character limit to text inputs

### DISCUSSION THREAD

**Discussion by developomp** [ppy/osu#21245] [Ideas] [answered]
When I try to send a lengthy message in the osu!lazer client, it clears the text input regardless of whether the message was successfully delivered or not, forcing me to retype the whole thing. Adding a character length limit feature and doing the following for message input area would feel like a much better user experience in my opinion:

1. When a user sends a message, lock the text input.
2. If the message is successfully delivered, clear the text input. Notify user otherwise (give the text input a little shake maybe).

**Comment by frenzibyte** [comment]
This is already how the text box behaves, where do you see this issue?

**Reply by developomp** [reply]
I clicked the message icon from a user's profile page in-game and I'm not sure if this is relevant, but this was my first time messaging them. The logs which I can't upload right now as I'm on my phone was complaining about https://osu.ppy.sh/api/v2/chat/new

**Comment by frenzibyte** [comment, marked as answer]
Tracking in https://github.com/ppy/osu/issues/21330.
